File: patterns/Strivers/Pattern18.java
public class Pattern18{
    public static void main(String[] args) {
          char end = (char)('A' + 5 - 1);  // chh = 'E'

        for (int i = 0; i < 5; i++) {
            char start = (char)(end - i);  // Explicit cast required

            for (char ch = start; ch <= end; ch++) {
                System.out.print(ch + " ");
            }
            System.out.println();
        }
    }
}

File: patterns/Strivers/Pattern18.cpp
#include<iostream>
using namespace std;


int main(){
    char end = 'A'+5-1; 
    for(int i =0 ; i<5 ;i++){
        int start = 'E'-i;
        for(char ch = start; ch<=end; ch++){
            cout<<ch<<" ";
        }
        cout<<endl;
    }


    return 0;
}